File: FrontEnd-ProjectReact/my-react-app-inventory/src/types/stockCountDetail.ts
import type { ItemDetail } from "./itemDetail";
import type { StockCountMaster } from "./stockCountMaster";

export interface StockCountDetail {
  id: number;
  scmno: string;
  itemDetailCode: string;
  countedQty?: number | null;
  recStatus?: string | null;
  itemDetailCodeNavigation: ItemDetail;
  scmnoNavigation: StockCountMaster;
}

File: Sample/Sample.Domain/Entities/StockCountDetail.cs
using Sample.Domain.Domain;
using System;
using System.Collections.Generic;

namespace Sample.Domain.Entities
{
    public class StockCountDetail : EntityBase, IAggregateRoot
    {
        public virtual int id { get; set; }
        public virtual string Scmno { get; set; } = null!;
        public virtual string ItemDetailCode { get; set; } = null!;
        public virtual int? CountedQty { get; set; }
        public virtual string? RecStatus { get; set; }

        public virtual ItemDetail ItemDetailCodeNavigation { get; set; } = null!;
        public virtual StockCountMaster ScmnoNavigation { get; set; } = null!;
    }
}
